src/sideloadedKit.ts

```diff
@@ -0,0 +1,68 @@
+import KitFilterHelper from './kitFilterHelper';
+import { IKitFilterSettings } from './configAPIClient';
+import { UnregisteredKit } from './forwarders.interfaces';
+import { EventTypeEnum, IdentityType } from './types.interfaces';
+
+/**
+ */
+export class MPSideloadedKit {
+    public kitInstance: UnregisteredKit;
+    public filterDictionary: IKitFilterSettings = {
+        eventTypeFilters: [],
+        eventNameFilters: [],
+        screenNameFilters: [],
+        screenAttributeFilters: [],
+        userIdentityFilters: [],
+        userAttributeFilters: [],
+        attributeFilters: [],
+        consentRegulationFilters: [],
+        consentRegulationPurposeFilters: [],
+        messageTypeFilters: [],
+        messageTypeStateFilters: [],
+        filteringEventAttributeValue: {},
+        filteringUserAttributeValue: {},
+        filteringConsentRuleValues: {},
+    } as IKitFilterSettings;
+
+    constructor(unregisteredKitInstance: UnregisteredKit) {
+        this.kitInstance = unregisteredKitInstance;
+    }
+
+    public addEventTypeFilter(eventType: EventTypeEnum) {
+        const hashedEventType  = KitFilterHelper.hashEventType(eventType);
+        this.filterDictionary.eventTypeFilters.push(hashedEventType);
+    }
+
+    public addEventNameFilter(eventType: EventTypeEnum, eventName: string) {
+        const hashedEventName  = KitFilterHelper.hashEventName(eventName, eventType);
+        this.filterDictionary.eventNameFilters.push(hashedEventName);
+    }
+    
+    public addEventAttributeFilter(eventType: EventTypeEnum, eventName: string, customAttributeKey: string) {
+        const hashedEventAttribute  = KitFilterHelper.hashEventAttributeKey(eventType, eventName, customAttributeKey);
+        this.filterDictionary.attributeFilters.push(hashedEventAttribute);
+    }
+
+    public addScreenNameFilter(screenName: string) {
+        const hashedScreenName = KitFilterHelper.hashEventName(screenName, EventTypeEnum.Unknown);
+        this.filterDictionary.screenNameFilters.push(hashedScreenName);
+    }
+
+    public addScreenAttributeFilter(screenName: string, screenAttribute: string) {
+        const hashedScreenAttribute = KitFilterHelper.hashEventAttributeKey(EventTypeEnum.Unknown, screenName, screenAttribute);
+        this.filterDictionary.screenAttributeFilters.push(hashedScreenAttribute);
+    }
+
+    public addUserIdentityFilter(userIdentity: IdentityType){
+        const hashedIdentityType = KitFilterHelper.hashUserIdentity(userIdentity);
+        this.filterDictionary.userIdentityFilters.push(hashedIdentityType);
+    }
+
+    public addUserAttributeFilter(userAttributeKey: string) {
+        const hashedUserAttributeKey = KitFilterHelper.hashUserAttribute(userAttributeKey);
+        this.filterDictionary.userAttributeFilters.push(hashedUserAttributeKey);
+    }
+
+    // public addCommerceEventAttributeFilter() {
```

Comment:
@einsteinx2 - I didn't see this as an array on the JS config, here is a sample. 
 https://jssdkcdns.mparticle.com/js/v2/e207c24e36a7a8478ba0fcb3707a616b/config

After updating the UI to add some ecommerce specific filters, I didn't see these pop up in any array filters after waiting for cache to bust either.  Maybe this is only an iOS/Android thing, (which would be a feature parity issue), but I'd like to see if anything pops up in the returned config for Android/iOS - I don't know how to get a sample one back though.  Maybe you can help...

Reply:
Looks like its actually a dictionary, keyed on the hash, with a 0 as the value if it's active. See our iOS unit test here: https://github.com/mParticle/mparticle-apple-sdk/blob/8ce7ad752ceeb1d9dc6d782d9c4462eb0ac114b9/UnitTests/MPKitContainerTests.m#L1257

I don't know if that's how it's done on Web though